File: app/src/main/java/mega/privacy/android/app/listeners/RemoveVersionsListener.java
package mega.privacy.android.app.listeners;

import android.content.Context;
import android.content.Intent;
import android.os.Handler;
import mega.privacy.android.app.MegaApplication;
import mega.privacy.android.app.R;
import mega.privacy.android.app.lollipop.ManagerActivityLollipop;
import mega.privacy.android.app.utils.Util;
import nz.mega.sdk.MegaApiJava;
import nz.mega.sdk.MegaError;
import nz.mega.sdk.MegaRequest;

import static mega.privacy.android.app.constants.BroadcastConstants.ACTION_RESET_VERSION_INFO_SETTING;

public class RemoveVersionsListener extends BaseListener {

    public RemoveVersionsListener(Context context) {
        super(context);
    }

    @Override
    public void onRequestFinish(MegaApiJava api, MegaRequest request, MegaError e) {
        super.onRequestFinish(api, request, e);
        if (context == null)
            return;
        switch (request.getType()) {
            case MegaRequest.TYPE_REMOVE_VERSIONS:
                if (e.getErrorCode() == MegaError.API_OK) {
                    Util.showSnackbar(context, context.getString(R.string.success_delete_versions));
                    MegaApplication.getInstance().sendBroadcast(new Intent(ACTION_RESET_VERSION_INFO_SETTING));
                    if (context instanceof ManagerActivityLollipop) {
                        final Handler handler = new Handler();
                        handler.postDelayed(() -> ((ManagerActivityLollipop) context).updateAccountStorageInfo(), 8000);
                    }
                } else {
                    Util.showSnackbar(context, context.getString(R.string.error_delete_versions));
                }
                break;
        }
    }
}

File: app/src/main/java/mega/privacy/android/app/listeners/RemoveVersionsListener.kt
package mega.privacy.android.app.listeners

import android.content.Context
import android.content.Intent
import android.os.Handler
import android.os.Looper
import mega.privacy.android.app.R
import mega.privacy.android.app.constants.BroadcastConstants
import mega.privacy.android.app.main.ManagerActivity
import mega.privacy.android.app.utils.Util
import nz.mega.sdk.MegaApiJava
import nz.mega.sdk.MegaError
import nz.mega.sdk.MegaRequest
import nz.mega.sdk.MegaRequestListenerInterface

/**
 * Listener to handle remove version api callback response
 *
 * @param context: Context
 */
class RemoveVersionsListener(private val context: Context) : MegaRequestListenerInterface {

    /**
     * Callback function for onRequestStart
     *
     * @param api : MegaApiJava
     * @param request : MegaRequest
     */
    override fun onRequestStart(api: MegaApiJava, request: MegaRequest) {
        // Do nothing
    }

    /**
     * Callback function for onRequestUpdate
     *
     * @param api : MegaApiJava
     * @param request : MegaRequest
     */
    override fun onRequestUpdate(api: MegaApiJava, request: MegaRequest) {
        // Do nothing
    }

    /**
     * Callback function for onRequestFinish
     *
     * @param api : MegaApiJava
     * @param request : MegaRequest
     * @param e: MegaError
     */
    override fun onRequestFinish(api: MegaApiJava, request: MegaRequest, e: MegaError) {
        if (request.type == MegaRequest.TYPE_REMOVE_VERSIONS) {
            if (e.errorCode == MegaError.API_OK) {
                Util.showSnackbar(context, context.getString(R.string.success_delete_versions))
                context.sendBroadcast(Intent(BroadcastConstants.ACTION_RESET_VERSION_INFO_SETTING))
            } else {
                Util.showSnackbar(context, context.getString(R.string.error_delete_versions))
            }
        }
    }

    /**
     * Callback function for onRequestTemporaryError
     *
     * @param api : MegaApiJava
     * @param request : MegaRequest
     * @param e: MegaError
     */
    override fun onRequestTemporaryError(api: MegaApiJava, request: MegaRequest, e: MegaError) {
        // Do nothing
    }
}